x-pack/plugins/security_solution/public/entity_analytics/components/entity_store/components/dashboard_panels.tsx

```diff
@@ -0,0 +1,246 @@
+
+  const entityStore = useEntityEngineStatus();
+  const riskEngineStatus = useRiskEngineStatus();
+
+  const { enable: enableStore } = useEntityStoreEnablement();
+  const { mutate: initRiskEngine } = useInitRiskEngineMutation();
+
+  const enableEntityStore = (enable: Enablements) => () => {
+    setModalState({ visible: false });
+    if (enable.riskScore) {
+      const options = {
+        onSuccess: () => {
+          setRiskEngineInitializing(false);
+          if (enable.entityStore) {
+            enableStore();
+          }
+        },
+      };
+      setRiskEngineInitializing(true);
+      initRiskEngine(undefined, options);
+    }
+
+    if (enable.entityStore) {
+      enableStore();
+    }
+  };
+
+  if (entityStore.status === 'loading') {
+    return (
+      <EuiPanel hasBorder>
+        <EuiEmptyPrompt
+          icon={<EuiLoadingSpinner size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_ENTITY_STORE}</h2>}
+        />
+      </EuiPanel>
+    );
+  }
+
+  if (entityStore.status === 'installing') {
+    return (
+      <EuiPanel hasBorder>
+        <EuiEmptyPrompt
+          icon={<EuiLoadingLogo logo="logoElastic" size="xl" />}
+          title={<h2>{ENABLEMENT_INITIALIZING_ENTITY_STORE}</h2>}
+          body={
+            <p>
+              <FormattedMessage
+                id="xpack.securitySolution.entityAnalytics.entityStore.enablement.initializing.description"
+                defaultMessage="It can take up to 5 minutes"
+              />
+            </p>
+          }
+        />
+      </EuiPanel>
+    );
+  }
+
+  const isRiskScoreAvailable =
+    riskEngineStatus.data &&
+    riskEngineStatus.data.risk_engine_status !== RiskEngineStatusEnum.NOT_INSTALLED;
+
+  return (
+    <EuiFlexGroup direction="column" data-test-subj="entityStorePanelsGroup">
+      {entityStore.status === 'enabled' && isRiskScoreAvailable && (
+        <>
+          <EuiFlexItem>
+            <EntityAnalyticsRiskScores riskEntity={RiskScoreEntity.user} />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EntityAnalyticsRiskScores riskEntity={RiskScoreEntity.host} />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EntitiesList />
+          </EuiFlexItem>
+        </>
+      )}
+      {entityStore.status === 'enabled' && !isRiskScoreAvailable && (
+        <>
+          <EuiFlexItem>
+            <EnableEntityStore
+              onEnable={() => setModalState({ visible: true })}
+              loadingRiskEngine={riskEngineInitializing}
+              enablements="riskScore"
+            />
+          </EuiFlexItem>
+
+          <EuiFlexItem>
+            <EntitiesList />
+          </EuiFlexItem>
+        </>
+      )}
+
+      {entityStore.status === 'not_installed' && !isRiskScoreAvailable && (
+        // TODO: Move modal inside EnableEntityStore component, eliminating the onEnable prop in favour of forwarding the riskScoreEnabled status
```

Comment:
Is this TODO for now or the future? If the future can you create a tech debt issue.

Reply:
yeah, for future I think. 
I have a bunch of tech debt issues to create. I will probably add them all tomorrow